src/wp-includes/class-wp-scripts.php

```diff
@@ -805,6 +792,10 @@ public function add_data( $handle, $key, $value ) {
 			return false;
 		}
 
+		if ( 'conditional' === $key ) {
+			$this->registered[ $handle ]->deps = array();
+		}
```

Comment:
Will you explain this? It seems to clear a script's dependencies if a conditional inline script is added, is that correct? Why is that desirable?

Reply:
Yes, that's exactly what it does. It handles the case where a script is only enqueued by a conditional script. If it's required by anything else, it will still be enqueued in that script's list of dependencies.

Reply:
👍 This makes sense to me now, I wasn't thinking about it correctly.

When a script is flagged as conditional, since it will never be printed, neither should its dependencies.

The alternative to clearing the dependencies here would be to build conditional logic into the dependencies class or to override more methods. With conditional scripts being deprecated, it seems like a good choice to make this small change instead of introducing more complicated logic elsewhere.

Reply:
I've added a comment to provide this with more context for the future.